oracle/src/oracleimporter.cpp

```diff
@@ -318,6 +322,26 @@ int OracleImporter::importTextSpoiler(CardSetPtr set, const QVariant &data)
     return cards;
 }
 
+void OracleImporter::sortColors(QStringList &colors)
+{
+    QMap<QString, unsigned int> colorOrder;
+    colorOrder["W"] = 0;
+    colorOrder["U"] = 1;
+    colorOrder["B"] = 2;
+    colorOrder["R"] = 3;
+    colorOrder["G"] = 4;
+    bool correctOrder = false;
+    while (!correctOrder) {
+        correctOrder = true;
+        for (unsigned int i = 0; i < colors.length() - 1; i++) {
```

Comment:
Do we really want to bubble sort this?

Reply:
Agreed

Reply:
I can't push it rn, but what about this solution?
```
QStringList sortColors(QStringList &colors) {
	QStringList orderedColors;
	QStringList colorOrder;
	colorOrder << "W" << "U" << "B" << "R" << "G";
	for (auto color : colorOrder) {
		int index = colors.indexOf(color);
		if (index != -1) {
			orderedColors << color;
			colors.removeAt(index);
		}
	}
	if (!colors.isEmpty()) {
		orderedColors.append(colors);
	}
	return orderedColors;
}
```
This was my original idea, but dismissed it because I thought bubble sort would be better when most of the multicoloured cards are maximum 2-3 colours, and sometimes in the correct colour order anyway.
Although I hadn't thought about the possibility of new colours, that would surely crush with the current code.

Reply:
Doesn't stl have sorting algorithms? Surely there must be a way where we don't have to implement anything more than a `(char, char) -> int` function

Reply:
https://www.codeproject.com/Articles/38381/STL-Sort-Comparison-Function#cpplambda

Reply:
Thanks for the link, it's much nicer that way!